lib/Dialect/TritonGPU/Transforms/Pipeline.cpp

```diff
@@ -1630,10 +1639,17 @@ void PipelinePass::asyncLaunchDots(scf::ForOp forOp) {
 
   // 0. insert dot_wait after the last dot in the loop
   Value dot = dots.back();
+  for (auto e : allDots) {
+    if (dot != e) {
+      hasSyncDot = true;
+    }
+  }
   auto loc = dot.getLoc();
-  builder.setInsertionPointAfter(dot.getDefiningOp());
-  auto dotWait = builder.create<tt::nvidia_gpu::DotWaitOp>(loc, dots.size());
-
+  Operation *dotWait = nullptr;
+  if (!hasSyncDot && !hasWaitGroup) {
+    builder.setInsertionPointAfter(dot.getDefiningOp());
```

Comment:
In general, DotWaitOp is coupled with DotAsyncOp.  If `hasSyncDot=true`, it will not insert `DotWaitOp`, but it will replace `DotOp` with `DotAsyncOp`？

Reply:
And so as the outstanding `DotWaitOp`.